### PR TITLE
Field name standardization

### PR DESCRIPTION
**Pull request by tsellers-r7**
## Description
This PR adds field names (`os.vendor`, `service.product`, etc) to `bin/recog_standarize`. This should help us identify field name typos as well as be a bit more intentional when we introduce a new field.

## How Has This Been Tested?
`ruby bin/recog_standardize xml/*.xml`, rspec, etc.

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have updated the documentation accordingly (or changes are not required).
- [ ] I have added tests to cover my changes (or new tests are not required).
- [ ] All new and existing tests passed.

### CODE REVIEW

File: identifiers/README.md [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+or syntax for a given identifier is preferred, this should be implemented in the
+application through a mapping function.
 
 ## Lists
 
+### Fields
+
+`fields.txt` defines the various fields (`os.vendor`, etc.) used to assert
+information about a match.
+
```
Comment (tsellers-r7) [inline identifiers/README.md:20]: The above is the only real change to this file. The rest are just formatting.

File: xml/html_title.xml [hunk 2]
```diff
@@ -1838,22 +1838,22 @@
...
   <fingerprint pattern="^AVTECH Software, Inc\. - TemPageR\s+([a-zA-Z0-9/]+) - Real-Time">
     <description>AVTECH TepPageR</description>
-    <example hw.product="TemPageR 3E" hw.product.model="3E">AVTECH Software, Inc. - TemPageR 3E - Real-Time Temperature Monitor For IT &amp;amp; Facilities Environment Monitoring</example>
+    <example hw.product="TemPageR 3E" hw.model="3E">AVTECH Software, Inc. - TemPageR 3E - Real-Time Temperature Monitor For IT &amp;amp; Facilities Environment Monitoring</example>
     <param pos="0" name="hw.vendor" value="AVTECH"/>
     <param pos="0" name="hw.device" value="Monitoring"/>
-    <param pos="1" name="hw.product.model"/>
-    <param pos="0" name="hw.product" value="TemPageR {hw.product.model}"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="0" name="hw.product" value="TemPageR {hw.model}"/>
```
Comment (tsellers-r7) [inline xml/html_title.xml:1856]: @hdm  - RE: These two fingerprints - you added these a while back.  I've changed `hw.product.model` to `hw.model` to be in keeping with the use of `hw.model` used elsewhere. Any concerns here?

File: xml/html_title.xml [hunk 3]
```diff
@@ -2766,7 +2766,7 @@
     <param pos="0" name="hw.certainty" value="0.75"/>
     <param pos="0" name="os.vendor" value="Dell"/>
     <param pos="0" name="os.product" value="iDRAC Linux"/>
-    <param pos="0" name="os .certainty" value="0.75"/>
+    <param pos="0" name="os.certainty" value="0.75"/>
```
Comment (tsellers-r7) [inline xml/html_title.xml:2769]: Already paying off..

**Comment by tsellers-r7** [comment]
@hdm - in light of the `hw.vendor` typo you caught earlier I figured it'd be time to implement this to help catch such things.

**Comment by hdm** [comment]
Nice work!

**Pull request merged**